common/lib/dependabot/file_updaters/base.rb

```diff
@@ -40,6 +40,8 @@ def file_changed?(file)
       end
 
       def requirement_changed?(file, dependency)
+        return true if lock_file?(file.name)
+
```

Comment:
Partial review here. It looks like this common class method is changing for a single ecosystem which worries me as it might have unintended consequences in other package managers and make refactoring this harder in future when we've lost this context.

I think it would be better to re-implementing this method change in the terrraform lockfile updater instead.

Althought, is this change required? Looks like it might not be needed anymore now that you're updating the lockfile outside the loop that updates the terraform and terragrunt files.

Reply:
I'm also worried about the unintended side-effects of this, so reimplementing it in the terraform ecosystem sounds like the way to go.

The requirement_changed? function is called indirectly in FileUpdater through file_changed?. Without a terraform lockfile dependency the function returns false, so this change resolves that check.